backend/lcfs/web/api/transaction/session.py

```diff
@@ -0,0 +1,154 @@
+        This method returns a list of OrganizationBase objects.
+        The OrganizationBase objects contain the basic organization details,
+        including the organization type, organization status, and other relevant fields.
+        The pagination object is used to control the number of results returned
+        and the page number.
+        The filters object is used to filter the results based on specific criteria.
+        The OrganizationBase objects are returned in the order specified by the sortOrders object.
+        The total_count field is used to return the total number of organizations that match the filters.
+        The OrganizationBase objects are returned in the order specified by the sortOrders object.
+
+        Args:
+            pagination (PaginationRequestSchema, optional): The pagination object containing page and size information. Defaults to {}.
+
+        Returns:
+            List[OrganizationBase]: A list of OrganizationBase objects containing the basic organization details.
+            The total_count field is used to return the total number of organizations that match the filters.
+            The OrganizationBase objects are returned in the order specified by the sortOrders object.
+
+        Raises:
+            Exception: If any errors occur during the query execution.
+            ValueError: If the provided pagination object is invalid.
+        """
+        try:
+            offset = 0 if (pagination.page < 1) else (
+                pagination.page - 1) * pagination.size
+            limit = pagination.size
+
+            query = select(Transaction).options(
+                joinedload(Transaction.issuance_history_record).options(
+                    joinedload(IssuanceHistory.organization),
+                    joinedload(IssuanceHistory.issuance_status),
+                ),
+                joinedload(Transaction.transfer_history_record).options(
+                    joinedload(TransferHistory.to_organization),
+                    joinedload(TransferHistory.from_organization),
+                    joinedload(TransferHistory.transfer_status),
+                ),
+                joinedload(Transaction.transaction_type),
+            )
+            count_query = await self.session.execute(
+                select(func.count(distinct(Transaction.transaction_id)))
+            )
+
+            total_count = count_query.unique().scalar_one_or_none()
+
+            transaction_results = await self.session.execute(query.offset(offset).limit(limit))
+
+            results = transaction_results.scalars().unique().all()
+
+            return [
+                Transaction.model_validate(transaction) for transaction in results
+            ], total_count
+
+        except Exception as e:
+            logger.error(f"Error occurred while fetching transactions: {e}")
+            raise Exception(f"Error occurred while fetching transactions")
+
+    async def export_transactions(
```

Comment:
Should this part be moved to views.py?

Reply:
Thanks for the comment. I think these changes are from Kevin's branch. I'll try to rebase mine with his after his branch is merged.